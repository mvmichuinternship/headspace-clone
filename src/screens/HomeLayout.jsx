import Bluebar from "../components/homepage-components/Bluebar";
import Homepage from "../components/homepage-components/Homepage";
import Navbar from "../components/homepage-components/Navbar";

function HomeLayout() {
  return (
    <div className="h-screen w-[100%]">
      <Bluebar />
      <Navbar />
      <Homepage />
    </div>
  );
}
export default HomeLayout;
