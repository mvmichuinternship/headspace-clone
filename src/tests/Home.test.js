import React from "react";
import renderer from "react-test-renderer";
import HomeLayout from "../screens/HomeLayout";
import { MemoryRouter } from "react-router-dom";

describe("Jest Snapshot testing suite", () => {
  it("Matches DOM Snapshot", () => {
    const domTree = renderer.create(<MemoryRouter><HomeLayout /></MemoryRouter>).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});