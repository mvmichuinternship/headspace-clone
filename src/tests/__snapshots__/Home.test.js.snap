// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="h-screen w-[100%]"
>
  <div
    className="flex justify-center items-center bg-sky-500 h-16 w-full"
  >
    <a
      className="underline text-white"
      href="/"
    >
      Get 14 day free trial
    </a>
  </div>
  <div
    className="flex justify-between w-full h-[10%] px-4 py-4  items-center"
  >
    <div
      className="flex w-[70%] gap-x-10 justify-start items-center"
    >
      <div
        className="md:hidden flex flex-col justify-center items-center"
      >
        <button
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </button>
      </div>
      <div
        className="flex justify-between items-center gap-x-3"
      >
        <img
          alt="Logo"
          className="w-8"
          src="logo1.png"
        />
        <span
          className="text-md"
        >
          headspace
        </span>
      </div>
      <div
        className="md:flex md:flex-row w-auto md:justify-around md:gap-x-6 hidden md:block"
      >
        <p
          className="text-md hover:font-bold"
        >
          For You
        </p>
        <p
          className="text-md hover:font-bold"
        >
          For business
        </p>
        <p
          className="text-md hover:font-bold"
        >
          Our plans
        </p>
        <p
          className="text-md hover:font-bold"
        >
          Resources
        </p>
        <p
          className="text-md hover:font-bold"
        >
          About
        </p>
      </div>
    </div>
    <div
      className="flex w-[50%] justify-end"
    >
      <div
        className="md:flex md:flex-row w-auto md:justify-around items-center md:gap-x-6 "
      >
        <p
          className="text-md hover:font-bold hidden md:block"
        >
          Help
        </p>
        <a
          className="text-md hover:font-bold cursor-pointer hidden md:block"
          href="/meditate"
          onClick={[Function]}
        >
          My Headspace
        </a>
        <div
          className="px-2 py-1 flex gap-x-1 text-center justify-center items-center rounded-full md:w-fit w-full md:px-5 px-2 py-4 bg-blue-600 text-white font-bold"
          data-testid="button"
        >
          <button>
            Try for free
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="h-full bg-stone-50 md:flex md:flex-row flex flex-col w-full md:justify-between justify-center items-center px-2 gap-x-10"
  >
    <img
      alt=""
      className="md:h-[525px] md:w-[596px] hidden md:block"
      src="https://images.ctfassets.net/v3n26e09qg2r/4D9CSEOzzm4zjCdPbmkFKE/927da3b4c128c841bb0abb4a16747f9e/Hero_Image_Left.webp?fm=webp&w=3840&q=75"
    />
    <img
      alt=""
      className="h-[592px] w-[492px] md:hidden block"
      src="https://images.ctfassets.net/v3n26e09qg2r/4NqNkoUd0hvoIvsRJrm2bS/50490f004b2a68e044e305e25c0da38a/Hero_Mobile_Image.webp?fm=webp&w=3840&q=75"
    />
    <div
      className="flex flex-col w-full justify-center items-center gap-y-8"
    >
      <span
        className="md:text-6xl text-4xl font-bold text-stone-700 leading-tight"
      >
        Make it your year with Headspace
      </span>
      <div
        className="w-full  w-full"
        data-testid="card"
      >
        <div
          className="w-full flex flex-col bg-yellow-400 rounded-lg px-3 py-7 md:px-10 "
        >
          <span
            className="font-extrabold text-stone-800 text-lg"
          >
            Free for 14 days, then ₹1499/year
          </span>
          <span
            className=""
          >
            ₹124.92 per month, billed annually
          </span>
        </div>
      </div>
      <div
        className=" font-light text-xs text-center flex gap-x-2 text-stone-700"
      >
        <span>
          After your free trial, the annual subscription is ₹1499 INR and automatically renews each year. Are you a post-secondary or university student? Save 85%, with valid SheerID student verification, for our Student plan
        </span>
      </div>
      <div
        className="px-2 py-1 flex gap-x-1 text-center justify-center items-center rounded-full md:w-fit w-[55%] px-5 py-4 bg-blue-600 text-white font-bold"
        data-testid="button"
      >
        <button>
          Try for free
        </button>
      </div>
    </div>
    <img
      alt=""
      className="md:h-[525px] md:w-[596px] hidden md:inline"
      src="https://images.ctfassets.net/v3n26e09qg2r/3Ciub8Qc1UhiOVboAfmazR/0a0737be4b3eabc7d9c9c0d00e9f94f7/Hero_Right_Image_Default-2022.png?fm=webp&w=3840&q=75"
    />
  </div>
</div>
`;
